mmv1/templates/terraform/examples/dns_managed_zone_private_multiproject.tf.erb

```diff
@@ -0,0 +1,124 @@
+    google_project_service.compute_project_1,
+    google_project_service.dns_project_1,
+    google_project_service.compute_project_2,
+    google_project_service.dns_project_2,
+  ]
+}
+
+resource "google_project" "project_1" {
+  name            = "<%= ctx[:vars]['project_1_name'] %>"
+  project_id      = "<%= ctx[:vars]['project_1_name'] %>"
+  org_id          = "<%= ctx[:test_env_vars]['org_id'] %>"
+  billing_account = "<%= ctx[:test_env_vars]['billing_account'] %>"
+}
+
+resource "google_project" "project_2" {
+  name            = "<%= ctx[:vars]['project_2_name'] %>"
+  project_id      = "<%= ctx[:vars]['project_2_name'] %>"
+  org_id          = "<%= ctx[:test_env_vars]['org_id'] %>"
+  billing_account = "<%= ctx[:test_env_vars]['billing_account'] %>"
+}
+
+resource "google_compute_network" "network_1_project_1" {
+  name                    = "<%= ctx[:vars]['network_1_name'] %>"
+  project                 = google_project.project_1.project_id
+  auto_create_subnetworks = false
+  depends_on              = [ 
+    google_project_service.compute_project_1,
+    google_project_service.dns_project_1,
+  ]
+}
+
+resource "google_compute_network" "network_2_project_1" {
+  name                    = "<%= ctx[:vars]['network_2_name'] %>"
+  project                 = google_project.project_1.project_id
+  auto_create_subnetworks = false
+  depends_on              = [ 
+    google_project_service.compute_project_1,
+    google_project_service.dns_project_1,
+  ]
+}
+
+resource "google_compute_network" "network_1_project_2" {
+  name                    = "<%= ctx[:vars]['network_1_name'] %>"
+  project                 = google_project.project_2.project_id
+  auto_create_subnetworks = false
+  depends_on              = [ 
+    google_project_service.compute_project_2,
+    google_project_service.dns_project_2,
+  ]
+}
+
+resource "google_compute_network" "network_2_project_2" {
+  name                    = "<%= ctx[:vars]['network_2_name'] %>"
+  project                 = google_project.project_2.project_id
+  auto_create_subnetworks = false
+  depends_on              = [ 
+    google_project_service.compute_project_2,
+    google_project_service.dns_project_2,
+  ]
+}
+
+resource "google_project_service" "compute_project_1" {
+  project = google_project.project_1.project_id
+  service = "compute.googleapis.com"
+  depends_on              = [
+    google_project.project_1,
+    google_project.project_2
+  ]
+}
+
+resource "google_project_service" "compute_project_2" {
+  project = google_project.project_2.project_id
+  service = "compute.googleapis.com"
+  depends_on              = [
+    google_project.project_1,
+    google_project.project_2
+  ]
+}
+
+resource "google_project_service" "dns_project_1" {
+  project = google_project.project_1.project_id
+  service = "dns.googleapis.com"
+  depends_on              = [
+    google_project.project_1,
+    google_project_service.compute_project_1,
+    google_project_service.dns_project_1,
```

Comment:
I think this won't work since it would give the resource a dependency on itself.

I'm pretty sure the dependency graph you'll need here & in the other example is:

- google_project_service.compute_project_2 _depends_on_ google_project_service.dns_project_1 and google_project_service.compute_project_1
- google_project_service.dns_project_2 _depends_on_ google_project_service.dns_project_1 and google_project_service.compute_project_1

(It looks like there are other tests in the providers already that have multiple `google_project_service` resources in a single project, so I think it would be okay to not have the `google_project_service.*_project_1` resources depend on both projects - but it also probably doesn't hurt.)

Reply:
Yeah, I agree with you and I have fixed that.

By the way, the dependency on itself was your suggestion [there](https://github.com/GoogleCloudPlatform/magic-modules/pull/8716#discussion_r1310652197) and I think by mistake 😄 
Anyway, I have tested an _itself_ dependency in a data resource and it worked. I have removed the dependency since it also for me makes no sense.

Reply:
whoops, sorry for the confusion 😂  Thanks!